app/vmselect/main.go

```diff
@@ -45,9 +45,7 @@ var slowQueries = metrics.NewCounter(`vm_slow_queries_total`)
 
 func getDefaultMaxConcurrentRequests() int {
 	n := cgroup.AvailableCPUs()
-	if n <= 4 {
-		n *= 2
-	}
+	n += min(n, 4)
```

Comment:
How about leaving just the following line instead of this line? It looks simpler and easier to reason about:

```go
n *= 2
```

Reply:
@valyala then for 12 cores it will be 24, ist it?

Reply:
> How about leaving just the following line instead of this line? It looks simpler and easier to reason about:
> 
> ```go
> n *= 2
> ```

Sure. I thought there was a reason why this condition was there
```go
if n <= 4 {
	n *= 2
}
```
That's why I left the multiplication under the original condition.
Changed!



> @valyala then for 12 cores it will be 24, ist it?

The upper limit (16 requests) hasn't been changed.

Reply:
> The upper limit (16 requests) hasn't been changed.

Oh, sure, you are correct. My bad.